src/main/webapp/clusters-view/clusters-view.controller.js

```diff
@@ -12,9 +12,60 @@ angular.module('managementConsole')
             $scope.group = modelController.getServer().getServerGroupName();
             $scope.stack = modelController.getServer().getFirstServer().getDefaultStack();
             $scope.relays = {};
+            $scope.offlineSites = {};
+            $scope.onlineSites  = {};
+            $scope.mixedSites   = {};
+
+            //
+            // Updates the map of remote site status
+            //
+            $scope.refreshRemoteSitesStatus = function(cluster) {
+              var resourcePathCacheContainer = cluster.domain.getFirstServer().getResourcePath()
+                 .concat('subsystem', 'datagrid-infinispan', 'cache-container', cluster.name);
+
+                // Refresh list of offline sites
+                cluster.modelController.readAttribute(resourcePathCacheContainer, 'offline-sites').then(
+                  function (response){
+                      if( response != null && response.constructor === Array) {
```

Comment:
No need for response != null check - I think. Because if this call succeeds then response has to be valid. Try it.